// @distube/file/dist/index.mjs
var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });

// src/index.ts
import { existsSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { DisTubeError, PlayableExtractorPlugin, Song } from "distube";
var FilePlugin = class extends PlayableExtractorPlugin {
  static {
    __name(this, "FilePlugin");
  }
  validate(url) {
    try {
      const u = new URL(url);
      if (u.protocol === "file:") return true;
    } catch {
    }
    return false;
  }
  resolve(url, options = {}) {
    const u = new URL(url);
    const path = fileURLToPath(u);
    const name = path.match(/[^/\\]*$/)?.[0] ?? path;
    if (!existsSync(path)) throw new DisTubeError("FILE_NOT_FOUND", `File not found: ${path}`);
    return new Song({ name, url, source: "file", playFromSource: true, plugin: this, id: path }, options);
  }
  getStreamURL(song) {
    if (!song.url) {
      throw new DisTubeError("FILE_PLUGIN_INVALID_SONG", "Cannot get stream url from invalid song.");
    }
    return song.url;
  }
  getRelatedSongs() {
    return [];
  }
};
export {
  FilePlugin
};
